### DISCUSSION TITLE
2024-07-12 - Product Backlog Refinement - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by bradystroud** [SSWConsulting/SSW.Rules.Content#8880] [Chewing the Fat]
https://www.ssw.com.au/rules/backlog-refinement-meeting

Backlog Refinement Meetings - have you done these and who should be required?

**Comment by Freego1783** [comment]
I've done some of those meetings in the past, it's really useful since projects tend to change along the way. Usually doing this with the dev team and calling in the PO if we need some clarification.

**Comment by tiagov8** [comment]
Most times PBIs can be refined without recurrent meetings, but I guess it dependes on the project.

The most important thing is to make sure any PBI is well defined enough to be completed without any deviation

**Comment by matt-goldman-ssw** [comment]
Backlog refinement is a game changer and I do it every Sprint now. Attendees are the Scrum Master and Product owner. You can pull other people in as required.

Video content from Calum was awesome but we need to get rid of that intro before we ship it. Also no mention in the video or rule about ordering the backlog - the refinement session is important to ensure the backlog reflects the PO's priority as well as ensuring that PBIs are Ready.

**Comment by danielmackay** [comment]
When I first started at SSW I asked about Refinement meetings, and these didn't seem to be happening.  I'm glad to see the are becoming more common place.  

Without Refinement meetings, I've witnessed:

- I've seen Sprint Planning turn into meetings of epic length
- Many PBIs get insta blocked during the Sprint, wasting lots of time

Personally, in Refinement I like to see primarily Tech Lead + Developers, which Product Owners being bought in as needed.  That way you can draw on the knowledge of everyone in the team, and developers can start thinking about the specifics of PBIs prior to actually working on them.

**Reply by wicksipedia** [reply]
> Many PBIs get insta blocked during the Sprint, wasting lots of time

Sounds like no one was reading the PBIs that they were pulling into the Sprint during the planning session 👀

**Comment by GabbyWard** [comment]
My current project team haven't done this, but when our last sprint planning meeting went overtime (2x the allocation) and it's because we are having to re-define the PBIs I think this would be a great initiative!

**Comment by ncn-ssw** [comment]
Though I can see there being value in such meetings for complex projects, there is no substitute for training your team to write PBIs properly in the first place.

**Comment by jakebayliss** [comment]
These are extremely important meetings and ideally I would do them every second sprint. Too often and you end up with a 10 minute meeting where you refined only a few PBI's.

I also dont think it is absolutely necessary that a PBI is 100% defined, refined, designed, estimated, ready at the time you pick it up, its not that hard to call the PO and ask them a question or 2. You will likely find yourself doing this even if the PBI is 'ready'.
 
I think it depends on the type of PO you have whether you have them in the refinement meeting. If they are technical and can add value when estimating or fleshing out PBI's then fine, but if not it is a waste of time for them and can be called in when they are needed for requirements

**Comment by ozdemir-mehmet** [comment]
These critical meetings should happen every second sprint to avoid short, unproductive sessions, and it's fine if a PBI isn't fully defined at the start, as you can always contact the PO for clarity; involving the PO in refinement meetings depends on their technical expertise and the value they add.

Hey I work in a steel mill

**Comment by YalalCakirSSW** [comment]
> Backlog Refinement Meetings - have you done these and who should be required?

Backlog Refinement is very important and necessary to ensure well defined PBIs and for smooth sprint planning experience. I think it should be done at least every 2-3 sprints. I haven't attended one yet. Tech lead and a dev would be sufficient.

**Comment by stephanfakossw** [comment]
N/a - not a dev (but I can see how this topic is important for the devs)

**Comment by leahy268** [comment]
Nothing worse than getting into Sprint Planning and then looking at the PBI and needing to spend heaps of time refining the items.  Super important to get that job over and done with before that point.

**Comment by jeoffreyfischer** [comment]
Good reminder as Backlog Refinement is often forgotten but is such a crucial phase of the Scrum process.

**Comment by tkapa** [comment]
The suggested approach of just incrementally defining PBIs rather than taking on the whole task is great. Sometimes this meeting can drag depending on the scope.

**Reply by SamBWagner** [reply]
I routinely find that these meetings are just total wastes of time as the refined items are de-prioritised and move to later on. 

It's better to have meetings with the Product Owner about stuff they're actively interested in doing and then just refining them on-the-fly in the sprint.

**Comment by Tanya-Leahy** [comment]
No, I haven't yet been involved in a Backlog Refinement Meeting. I have come across quite a few PBIs that aren't refined enough to work on without further information. At least a tech lead and developer should be involved in these meetings.

**Comment by isaaclombardssw** [comment]
I haven't come across these meetings before, but I have come across PBIs warranting refinement. Moving these refinement tasks out of the team-wide meetings would have saved some time.

**Comment by Calinator444** [comment]
Product Backlog refinement is a great for ensuring that work done during the sprint is well defined so that there are few hold ups or debates mid sprint.

**Comment by uly1** [comment]
IMO it shouold be the Tech Lead and the PO, and they bring in others if needed.
Also this might only really make sense for larger teams... if there's only 2 devs, just combine it with the planning meeting since you're using the same amount of people's time anyway and there's less context switching

**Comment by joshbermanssw** [comment]
I find these quite useful - especially as of late. These meetings can have as little as the PO and Tech Lead, all the way to the full team. Ultimately depending on the state of the PBIs in the backlog and who is asking to sit in on them eeting.

**Comment by bahjatalaadelssw** [comment]
Sometimes we run them before Sprint Reviews because in most cases the TV over commits in a 1 week sprint.

**Comment by awaltos** [comment]
I have been in many refinement meetings, and I always found them to be valuable. I think a PO + a technical person needs to be there as a minimum. Tech people can often identify edge cases or potential blockers and PO or BA need to clarify/edit ACs and approve any changes. Others such as Test lead or SA can be brought in ad hoc.

**Comment by steven0x51** [comment]
My current project hasn't had Backlog Refinement Meeting. I think Teck Lead and developer are required for this meeting

**Comment by Dhruv-0987** [comment]
Backlog Refinement Meetings - have you done these and who should be required?

I have not been in many but thats because it is more efficient for just the product owner, tech lead and scrum master to do it while other people continue on with their work, if need be anyone can be pulled to ask specific questions

**Comment by jimmy1992abc** [comment]
For our accounting team, our backlog won't be too much, and most of them are recurring. It is much easier for us to skip the details on the recurring and go through other PBIs if we need something different. Ideally every team members will be involved.

**Comment by miichaelsmedley** [comment]
A good product owner should be all over this but work with someone technical to define any limitations.

**Comment by PennyWalker** [comment]
This has got to be a time saver - the last thing you want is a team of 10 people all sitting and refining during planning sessions. I think the tech lead and another team member could flesh out a lot of them, but I suspect the PO would be needed a fair amount if the requirements aren't clear.

**Comment by MarcusIrmscherSSW** [comment]
Great video Calum, nice concise explanations. 🌠

**Comment by cvionnet** [comment]
> who should be required?

Tech Lead and Devs.
Then, it depends. If the team has no idea about the top 5 PBIs, then the PO should be there. Otherwise, it's fine to include him only for specific requirements.

Agree with @ncn-ssw . The better defined the PBIs are when created, the less time is needed in the meeting.

**Comment by ricksu978** [comment]
PO, tech lead and the developers should attend the refinement meeting.

How often should you have the refinement meeting?
It'll depend on how quick the developers can close the refined PBIs.
The quicker the more refinement meetings you'll need.

**Comment by bettybondoc** [comment]
In the Design team, we operate a bit differently since we're not a Scrum team dedicated to a single project. We receive various design requests from multiple teams (TV, Marketing, and our numerous internal projects). While we have our own design PO, we also collaborate with several other POs/Product Champions/devs who have their own urgent tasks.

As a result, we often overcommit and add more tasks mid-sprint, but that's another conversation.

> Backlog Refinement Meetings - have you done these and who should be required?

Typically, people send us requests, which we turn into PBIs and usually assign to ourselves. If needed, we hand them over to someone else. Therefore, the PBIs are usually well-defined before we pick them up, as we flesh out the details from the beginning. We've done backlog refinement once, but separately—each designer was assigned to review their own PBIs (and a couple of unassigned ones) and refine, remove, or reassign them.

Ideally, I would like a design lead (or tech lead) and a PO to handle backlog refinement for us occasionally because it's quite challenging to pick a PBI from a sea of urgent requests.

**Comment by Ben0189** [comment]
Backlog refinement meetings are important and highly valuable. It make sure that developers who might be picking up tasks have a clear understanding of what they need to do, and the Product Owner knows what to expect when the PBI is completed. These meetings should involve the tech lead, developers, and the PO.

**Comment by amankumarrr** [comment]
For client work, especially when you don't have extensive knowledge about their business domain, it is crucial to hold Refinement Meetings. These meetings provide an opportunity to understand details that might not be captured in the PBIs, share information among the team, and gain clarity before starting work. Discussing PBIs with Product Owners ensures that expectations are clear and that any potential misunderstandings are addressed.

Refinement Meetings help achieve consensus among the entire team, including the PO, Tech Lead, and developers, on the objectives and requirements of each PBI. This alignment is essential for the team's success.

**Comment by chrisschultzssw** [comment]
N/A, not a dev :). Well-defined PBIs are super important though.

Great video 🪠

**Comment by AttackOnMorty** [comment]
> who should be required?

PO, TL and at least one Dev. 

PO should be in the Refinement meeting, they can:
- Provide clarity on the requirements
- Prioritize PBIs
- Know what is refined

It will make the Refinement meeting and subsequent Planning meeting more efficient.

**Comment by christoment** [comment]
> Backlog Refinement Meetings - have you done these and who should be required?

TL and developers for sure, but PO and scrum master should be in the refinement meeting too so we can order the backlog according to the PO's priority 😄

**Comment by AntPolkanov** [comment]
> have you done these 

Some time ago we started doing backlog refinements for SSW Rewards. It helped not only to make our Sprints easier, but also drop many duplicate or not relevant tickets.

> who should be required?

Product owner, scrum master and tech lead.

**Comment by duanxinhuan** [comment]
JUST USE YAKSHAVER GUYS.
It solves all the problems

**Reply by duanxinhuan** [reply]
You shouldn't be marketing a product that is not ready Jack!

**Reply by duanxinhuan** [reply]
Sorry jack

**Reply by duanxinhuan** [reply]
I forgive youo

**Comment by MattParkerDev** [comment]
> Backlog Refinement Meetings - have you done these

Yes

> and who should be required?

Tech lead and a developer, PO and SM

**Comment by will-truscott-ssw** [comment]
Personally, I feel backlog refinement meetings should occur whenever an issue is identified to avoid unnecessary personnel costs and meeting fatigue from your product owner. For example, if you've noticed any times where a dev has been unsure about a task, it's likely you need a refinement meeting! In my experience this generally happens every 3 or so sprints as requirements evolve.

**Comment by SamBWagner** [comment]
> Backlog Refinement Meetings - have you done these and who should be required?

I think they're a good tool to add to the tool-belt. However, they're not a silver bullet and they can lead to an **extreme** level of time wasting. 

I think this should be deployed team-to-team and pbi-to-pbi. If a large scale refinement meeting is what works best for your team, and you can prove that, then that's awesome! 

If you're better off doing it on-the-fly, then that's also great!

Your team should define the process. The process should **not** define how the team works.

**Comment by jaydenalchin** [comment]
Frequent backlog refinements are super important - refined, understandable, prioritised PBIs go a long way. However, I don't think the whole team of developers need to be involved in them. A backlog refinement session should really just be a Product Owner refining and prioritising items in the Product Backlog and pulling in the team members they need to support that as they go.

**Comment by jimmidier** [comment]
> have you done these and who should be required

I haven't but have seen the outcome of the refinement meeting where only the PO and the scrum master took part in to avoid consuming too much time for the rest. PO should be required for sure, with one of the other developers for technical information.

**Comment by JerryLuo5799** [comment]
> Backlog Refinement Meetings - have you done these and who should be required?

I think it's very important, in my current client project, we need tech leader, a dev, and PO, we think PO should confirm all the refinement .

**Comment by Aibono1225** [comment]
> Backlog Refinement Meetings - have you done these and who should be required?

No, I haven't. I think Tech Lead, Devs and PO should be required.

**Comment by KristenHu** [comment]
> Backlog Refinement Meetings - have you done these and who should be required?

No, but we've re-defined PBIs in sprint planning sometimes. Tech Lead and PO must be required.

**Comment by babakamyljanovssw** [comment]
Before Sprint, usually PO or Scrum Master add tag to PBI e.g."Sprint xxx", that will help to do Sprint Planning smoothly

**Comment by tino-liu** [comment]
Backlog Refinement is great to have, but an already well written PBI and transparency culture does make it less required.

**Comment by JeanThirion** [comment]
> Backlog Refinement Meetings - have you done these and who should be required?

Big projects with many devs ? Regardless of whether it's an appointment or not, this should be the scrum masters' priority to make sure the backlog is always as refined as possible, with the help of the tech lead.

Should it be a recurring appointment? Well, if you can't get it done "naturally" then yeah probably.

Smaller scale project ? (I.e. typical 2 devs + SM) 
I don't think it's worth having a recurring meeting, you should be able to manage a clean backlog naturally. But if you can't then again, that's a tool.

**Comment by drwharris** [comment]
I dont know how anyone can go into planning without a fair degree of refinement done. 

I find refinement meetings that have the whole team go off track rapidly and waste the time of a fair amount of the team.

My ideal refinement meeting deals with one Feature, it includes the Tech Lead, 1 SME developer, 1 Tester, the PO and SM.

**Comment by JackDevAU** [comment]
Definitely need a refinement meeting or the "planning" becomes it...

**Comment by TobyChurches** [comment]
> Backlog Refinement Meetings - have you done these and who should be required?

I have but it is generally Ad Hoc, There isn't a set recurrence, but it usually involves a Dev and the tech lead.

**Comment by vladislav-kir** [comment]
> Backlog Refinement Meetings - have you done these and who should be required?

In my experience Backlog Refinement Meetings are extremely useful! On my current client, I do them almost every week with the Product Owner and one more dev.

**Comment by PothieuG** [comment]
Backlog Refinement is great, but it become less important if the project is small and PBIs are already well written.
